core/src/main/java/com/netflix/conductor/validations/WorkflowTaskTypeConstraint.java

```diff
@@ -0,0 +1,169 @@
+package com.netflix.conductor.validations;
+
+import com.netflix.conductor.common.metadata.workflow.TaskType;
+import com.netflix.conductor.common.metadata.workflow.WorkflowTask;
+
+import javax.validation.Constraint;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import javax.validation.Payload;
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.ElementType.TYPE;
+
+/**
+ * This constraint class validates following things.
+ * 1. Correct parameters are set depending on task type.
+ */
+@Documented
+@Constraint(validatedBy = WorkflowTaskTypeConstraint.WorkflowTaskValidator.class)
+@Target({TYPE,  ANNOTATION_TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface WorkflowTaskTypeConstraint {
+    String message() default "";
+
+    Class<?>[] groups() default {};
+
+    Class<? extends Payload>[] payload() default {};
+
+    class WorkflowTaskValidator implements ConstraintValidator<WorkflowTaskTypeConstraint, WorkflowTask> {
+
+        final String PARAM_REQUIRED_STRING_FORMAT = "%s field is required for taskType: %s taskName: %s";
+
+        @Override
+        public void initialize(WorkflowTaskTypeConstraint constraintAnnotation) {
+        }
+
+        @Override
+        public boolean isValid(WorkflowTask workflowTask, ConstraintValidatorContext context) {
+            context.disableDefaultConstraintViolation();
+
+            boolean valid = true;
+
+            // depending on task type check if required parameters are set or not
+            switch (workflowTask.getType()) {
+                case TaskType.TASK_TYPE_EVENT:
+                    try {
+                        valid = isEventTaskValid(workflowTask, context);
+                    } catch (NoSuchFieldException e) {
```

Comment:
It doesn't look like isEventTaskValid method actually throws NoSuchFieldException, also don't leave `e.printStackTrace()` in the code, logging instead.